packages/db/fauna/resources/Function/renameUserUpload.js

```diff
@@ -0,0 +1,66 @@
+import fauna from 'faunadb'
+
+const {
+  Abort,
+  Collection,
+  CreateFunction,
+  Exists,
+  Function,
+  Get,
+  If,
+  Index,
+  IsNonEmpty,
+  Lambda,
+  Let,
+  Match,
+  Query,
+  Ref,
+  Select,
+  Update,
+  Var
+} = fauna
+
+const name = 'renameUserUpload'
+const body = Query(
+  Lambda(
+    ['userId', 'cid', 'name'],
+    Let(
+      {
+        userRef: Ref(Collection('User'), Var('userId')),
+        contentMatch: Match(Index('unique_Content_cid'), Var('cid'))
+      },
+      If(
+        IsNonEmpty(Var('contentMatch')),
+        Let(
+          {
+            content: Get(Var('contentMatch'))
+          },
+          Let(
+            {
+              uploadMatch: Match(
+                Index('upload_by_user_and_content'),
+                Var('userRef'),
+                Select('ref', Var('content')),
+                true
+              )
+            },
+            If(IsNonEmpty(Var('uploadMatch')),
+              Update(
+                Select(['ref'], Get(Var('uploadMatch'))),
+                { data: { name: Var('name') } }
```

Comment:
Would be great if this set `updated` - maybe add to schema if not already?

Reply:
@alanshaw I avoided doing too much changes in the schema side of things because we're currently migrating to postgres, but I could do it now if needed